src/controllers/currentWarningsController.js

```diff
@@ -0,0 +1,155 @@
+/* eslint-disable */
+const mongoose = require('mongoose');
+const userProfile = require('../models/userProfile');
+
+const currentWarningsController = function (currentWarnings) {
+  const checkForDuplicates = (currentWarning, warnings) => {
+    const duplicateFound = warnings.some(
+      (warning) => warning.warningTitle.toLowerCase() === currentWarning,
+    );
+
+    return duplicateFound;
+  };
+
+  const checkIfSpecialCharacter = (warning) => {
+    return !/^[a-zA-Z][a-zA-Z0-9]*(?: [a-zA-Z0-9]+)*$/.test(warning);
+  };
+  const getCurrentWarnings = async (req, res) => {
+    try {
+      const response = await currentWarnings.find({});
+
+      if (response.length === 0) {
+        return res.status(400).send({ message: 'no valid records' });
+      }
+      return res.status(200).send({ currentWarningDescriptions: response });
+    } catch (error) {
+      res.status(401).send({ message: error.message || error });
+    }
+  };
+
+  const postNewWarningDescription = async (req, res) => {
+    try {
+      const { newWarning, activeWarning, isPermanent } = req.body;
+
+      const warnings = await currentWarnings.find({});
+
+      if (warnings.length === 0) {
+        return res.status(400).send({ message: 'no valid records' });
+      }
+
+      const testWarning = checkIfSpecialCharacter(newWarning);
+      if (testWarning) {
+        return res.status(200).send({
```

Comment:
Errors shouldn't be returning a `200` code, a generic `400` should be enough. And why is errors returning two different bodies? (`{ message: string }` and `{ error: string }`)

Reply:
Hello
Thanks for the feedback!
I send a 200 because if I send a 400 error, there is a red warning that appears in the console and wanted to avoid that. After reaching out team members this was the solution